hal/spi_api.h

```diff
@@ -33,6 +33,7 @@
 #define SPI_EVENT_INTERNAL_TRANSFER_COMPLETE (1 << 30) // Internal flag to report that an event occurred
 
 #define SPI_FILL_WORD         (0xFFFF)
+#define SPI_FILL_CHAR         (0xFF)
```

Comment:
Do we need this char? SPI supports 4-16 bit often, therefore we can use word, have word fill member to be 16-bit wide. 

Looking at SPI class, all write support up to 16 bit, only the latest addition does not (buffers are type of char*) ?